File: Anole/HTMLRenderTableItem.cs
using System;
using System.Drawing;

namespace Anole
{
	/// <summary>
	/// Summary description for HTMLRenderTableItem.
	/// </summary>
	public class HTMLRenderTableItem : HTMLRenderItem
	{
		protected int myRowCount;
		protected int myColCount;
		protected int[] myColWidths;
		protected int[] myRowHeights;
		
		HTMLTableElement myTableElement;

		public HTMLRenderTableItem(int rowcount,
			int colcount, int width, int height, HTMLTableElement element,
			int[] colwidths, int[] rowheights)
		{
			this.myHTMLElement=element;
			this.myTableElement=element;
			this.myRowCount=rowcount;
			this.myColCount=colcount;
			this.myWidth=width;
			this.myHeight=height;
			this.myColWidths=colwidths;
			this.myRowHeights=rowheights;
		}

		public override void Paint(Graphics g)
		{
			Pen p=new Pen(Color.Red);
			g.DrawRectangle(p,Left,Top,Width,Height);

			int l=this.Left;
			int t=this.Top;

			for(int i=0; i<myRowCount; i++)
			{
				for(int j=0; j<myColCount; j++)
				{
					g.DrawRectangle(p,l,t,this.myColWidths[j],this.myRowHeights[i]);
					l+=this.myColWidths[j];
				}
				l=this.Left;
				t+=this.myRowHeights[i];
				
			}

		}

			//g.DrawLine(p,this.Left,this.Top,this.Width,this.Top);
		}
	}

File: Anole/HTMLRenderTableItem.vb
Imports System
Imports System.Drawing

Namespace Anole
    ''' <summary>
    ''' Summary description for HTMLRenderTableItem.
    ''' </summary>
    Public Class HTMLRenderTableItem
        Inherits HTMLRenderItem
        Protected myRowCount As Integer
        Protected myColCount As Integer
        Protected myColWidths As Integer()
        Protected myRowHeights As Integer()
        
        Private myTableElement As HTMLTableElement
        
        Public Sub New(rowcount As Integer, colcount As Integer, width As Integer, height As Integer, element As HTMLTableElement, colwidths As Integer(), _
        	rowheights As Integer())
            Me.myHTMLElement = element
            Me.myTableElement = element
            Me.myRowCount = rowcount
            Me.myColCount = colcount
            Me.myWidth = width
            Me.myHeight = height
            Me.myColWidths = colwidths
            Me.myRowHeights = rowheights
        End Sub
        
        Public Overloads Overrides Sub Paint(g As Graphics)
            Dim p As New Pen(Color.Red)
            g.DrawRectangle(p, Left, Top, Width, Height)
            
            Dim l As Integer = Me.Left
            Dim t As Integer = Me.Top
            
            For i As Integer = 0 To myRowCount - 1
                For j As Integer = 0 To myColCount - 1
                    g.DrawRectangle(p, l, t, Me.myColWidths(j), Me.myRowHeights(i))
                    l += Me.myColWidths(j)
                Next
                l = Me.Left
                    
                t += Me.myRowHeights(i)
            Next
            
        End Sub
        
        'g.DrawLine(p,this.Left,this.Top,this.Width,this.Top);
    End Class
End Namespace
